test/services/DataGeneratorSpec.scala

```diff
@@ -195,4 +198,89 @@ class DataGeneratorSpec extends PlaySpec with WithFakeApplication with ErsTestHe
+      val failedValue: Either[Throwable, DataValidator] = testDataGen.setValidatorCsv("CSOP_OptionsGranted_V3")
+
+      val returnedExceptionExample: ERSFileProcessingException = ERSFileProcessingException(
+        "ers.exceptions.dataParser.configFailure",
+        Messages("ers.exceptions.dataParser.validatorError"),
+        optionalParams = Seq("CSOP_OptionsGranted_V3")
+      )
+
+      assert(failedValue.isLeft)
+      assert(failedValue.left.get.isInstanceOf[ERSFileProcessingException])
+      failedValue.left.get.asInstanceOf[ERSFileProcessingException].context mustEqual returnedExceptionExample.context
+      failedValue.left.get.asInstanceOf[ERSFileProcessingException].message mustEqual returnedExceptionExample.message
+      failedValue.left.get.asInstanceOf[ERSFileProcessingException].optionalParams mustEqual returnedExceptionExample.optionalParams
+
+      verify(mockAuditEvents, times(1))
+        .auditRunTimeError(
+          ArgumentMatchers.eq(thrownException),
+          ArgumentMatchers.eq("Could not set the validator"),
+          ArgumentMatchers.eq("CSOP_OptionsGranted_V3"))(any(), any(), any())
+    }
+  }
+
+  "identifyAndDefineSheetEither" must {
+    "return Right with the sheet name if given valid input" in new DataGeneratorObj("CSOP") {
+      identifyAndDefineSheetEither((SheetInfo("someInput",1,"anInput","","",List("")), "SOMEINPUT")) mustBe Right("anInput")
+    }
+
+    "return Left with a processing exception if given invalid input" in {
+      when(mockErsUtil.withArticle(any())).thenReturn("ersUtilReturn")
+      val testDataGen: DataGenerator = new DataGenerator(mockAuditEvents, mockMetrics, testParserUtil, testErsValidationConfigs, mockErsUtil)
+
+      val failedValue: Either[Throwable, String] = testDataGen
+        .identifyAndDefineSheetEither((SheetInfo("differentInput",1,"aSheetName","","",List("")), "SOMEINPUT"))
+      val returnedExceptionExample: ERSFileProcessingException = ERSFileProcessingException(
+        "ers.exceptions.dataParser.incorrectSchemeType",
+        Messages("ers.exceptions.dataParser.incorrectSchemeType", "differentinput", "someinput"),
+        optionalParams = Seq("ersUtilReturn", "ersUtilReturn", "aSheetName")
+      )
+
+      assert(failedValue.isLeft)
+      assert(failedValue.left.get.isInstanceOf[ERSFileProcessingException])
+      failedValue.left.get.asInstanceOf[ERSFileProcessingException].context mustEqual returnedExceptionExample.context
+      failedValue.left.get.asInstanceOf[ERSFileProcessingException].message mustEqual returnedExceptionExample.message
+      failedValue.left.get.asInstanceOf[ERSFileProcessingException].optionalParams mustEqual returnedExceptionExample.optionalParams
+
+      verify(mockAuditEvents, times(1)).fileProcessingErrorAudit("differentInput", "aSheetName", "differentinput is not equal to someinput")
+
+
+    }
+  }
+
+  //TODO these are not strictly unit tests, they go out to ersSheets.
```

Comment:
Does this TODO need following up on?

Reply:
@DannyPennington97 do you reckon it does? I can't remember if it'd take a lot of effort to bring ersSheets into DataGenerator

Reply:
I'd say probably not? ersSheets is basically just a big map of strings so re-creating/mocking output for tests seems unnecessary to me